_articles/scripting/Typescript/typescript-modifier.md

```diff
@@ -0,0 +1,231 @@
+
+		// Fetch target
+		const target = this.GetCursorTarget()!;
+
+		// Play sound
+		target.EmitSound(this.sound_cast);
+
+		// Add modifier
+		target.AddNewModifier(this.GetCaster(), this, "modifier_typescript_ancient_seal", { duration: seal_duration });
+	}
+}
+```
+
+Great! This applies the modifier on the target. But we haven't defined the modifier yet, so let's do that next.
+
+### Creating The Modifier
+
+This part is absolutely up to you and your organizational preferences: some prefer to add the modifier as a separate file, while some prefer to have the ability and its associated modifiers in the same file. You could place the modifier file inside `src/vscripts/modifiers`, for instance. In order to keep the guide simple, let's make the modifier in the same file.
+
+Very similar to an ability in TS, modifiers are also a class. We create a modifier with the following structure:
+
+```ts
+@registerModifier()
+export class modifier_typescript_ancient_seal extends BaseModifier {
+
+}
+```
+
+As you can see, it's very similar to an ability, replacing `@registerAbility()` with `@registerModifier()`, and the `BaseAbility` extension with `BaseModifier`.
+Note that `@registerModifier()` takes care of LinkLuaModifier for you, so you don't need to call it on TS modifiers.
+
+### Typechecking Modifier Calls
+
+Before we continue, one thing we can do to add some more type checking is to link the ability to the modifier. Instead of relying on a string for the naming of the modifier, we'll instead link the class name.
+
+To do so, we simply remove the quotation marks around the modifier name, then add `.name` to it. See below the code before and after linking the class:
+
+<MultiCodeBlock titles="Before|After">
+
+```ts
+// Add modifier
+target.AddNewModifier(this.GetCaster(), this, "modifier_typescript_ancient_seal", { duration: seal_duration });
+```
+
+```ts
+// Add modifier
+target.AddNewModifier(this.GetCaster(), this, modifier_typescript_ancient_seal.name, { duration: seal_duration });
+```
+
+</MultiCodeBlock>
+
+This results at the exact name of the modifier as a string, which is enforced by Typescript.
+
+:::note
+If your modifier is in another file, you'll have to import it first before you can link it in the above fashion.
+:::
+
+### Coding The Modifier
+
+Alright. Let's set and apply some properties for the modifier, such as the parent, the ability and the particle effect. In addition, let's set some useful properties via modifier functions.
+Also, this is my personal choice, but I usually put ability specials as a class property so they can be easily used everywhere in the modifier.
+
+```ts
+@registerModifier()
+export class modifier_typescript_ancient_seal extends BaseModifier {
+	ability = this.GetAbility()!;
+	parent = this.GetParent();
+	particle_seal = "particles/units/heroes/hero_skywrath_mage/skywrath_mage_ancient_seal_debuff.vpcf";
+	resist_debuff?: number;
+
+	IsHidden() {
```

Comment:
Would be nice if you could add some comments inside each function with what they do, ie IsHidden `// If false is returned, this modifier shows on the player's HUD, otherwise it is hidden.`, OnCreated `// This method is called whenever a ew instance of this modifier is applied to a unit`

Reply:
Added comments.